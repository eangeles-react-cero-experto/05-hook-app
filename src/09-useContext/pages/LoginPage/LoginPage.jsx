import { useContext } from "react";
import { UserContext } from "../../context/UserContext.jsx";
import { FormLogin } from "../../components/index.js";

export const LoginPage = () => {
  const { user } = useContext(UserContext);

  return (
    <>
      <h1>LoginPage</h1>
      <hr />

      <FormLogin />

      <br />
      <br />

      <pre data-testid="code-block">{JSON.stringify(user, null, 4)}</pre>
    </>
  );
};
