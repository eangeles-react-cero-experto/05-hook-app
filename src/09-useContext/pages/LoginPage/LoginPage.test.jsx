import { render, screen } from "@testing-library/react";
import { LoginPage } from "./LoginPage.jsx";
import { UserContext } from "../../context/UserContext.jsx";

describe("Test LoginPage", () => {
  const user = {
    username: "John Due",
    email: "john@email.com",
  };

  it("should match with snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={{ user }}>
        <LoginPage />
      </UserContext.Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render the component without the user", () => {
    render(
      <UserContext.Provider value={{ user: null }}>
        <LoginPage />
      </UserContext.Provider>,
    );

    const preElement = screen.getByTestId("code-block");
    expect(preElement.innerHTML).toEqual("null");
  });

  it("should render the component with the user", () => {
    render(
      <UserContext.Provider value={{ user }}>
        <LoginPage />
      </UserContext.Provider>,
    );

    const preElement = screen.getByTestId("code-block");
    expect(preElement.innerHTML).toContain(user.username);
    expect(preElement.innerHTML).toContain(user.email);
  });
});
