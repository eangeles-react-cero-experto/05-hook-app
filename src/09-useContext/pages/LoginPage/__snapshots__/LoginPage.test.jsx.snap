// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginPage should match with snapshot 1`] = `
<div>
  <h1>
    LoginPage
  </h1>
  <hr />
  <form>
    <input
      class="form-control"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <br />
    <input
      class="form-control"
      name="email"
      placeholder="email"
      type="email"
      value=""
    />
    <br />
    <br />
    <button
      class="btn btn-primary"
      type="submit"
    >
        Login  
    </button>
  </form>
  <br />
  <br />
  <pre
    data-testid="code-block"
  >
    {
    "username": "John Due",
    "email": "john@email.com"
}
  </pre>
</div>
`;
